### DISCUSSION TITLE
Using cert-manager for two-way ssl

### DISCUSSION THREAD

**Discussion by N0tronic** [emqx/emqx#10840] [Q&A]
Hey guys,

I´ve a problem with the cert-manager usage. I want use them to automate cert creation. For the emqx Dashboard that working fine.
 
But I want use the generated certs also for the mqtt connection. But emqx ingress default config forward the entry port 8883 to the emqx mqtt port 1883. So I can´t use Two-Way SSL functions from emqx broker and don´t safe my mqtt connection with client certificates.

How can I solve this problem? For example, Is it possible create client certificates over cert-manager ander terminate Two-Way-SSL on the ingress-controller? Or is it the only possibility to use the cert manager without client cert authentication? 

Thx and cheers

**Comment by lafirest** [comment]
@N0tronic 
Hi, There are two facts I am sure of:

1. In EMQX we don't have a `cert-manager` feature
2. We never forward the connection to other port

Could you provide more information like your environment, EMQX version, configuration, and your logs?

**Reply by N0tronic** [reply]
I'm using EMQX 5.0.22

Look at helm deployment (ingress section):
deploy/charts/emqx/values.yaml

Helm deployment isn't a part of the EMQX application? There are resources for cert-manager integration.

**Reply by lafirest** [reply]
I guess maybe you are discussing the [EMQX Operator](https://github.com/emqx/emqx-operator)? It is another project.

**Reply by N0tronic** [reply]
Operator is not helm. Helm deployment is part of this EMQX repository. And there r mechanisms for ingress and acme challenge...

... confusing that nobody knows something about that 😁

**Reply by lafirest** [reply]
@id 
Could you have a look at this?

**Reply by thalesmg** [reply]
@Rory-Z would you know as well?
